starknet_py/tests/e2e/fixtures/contracts.py

```diff
@@ -155,30 +158,42 @@ def fixture_balance_contract() -> str:
     return read_contract("balance_compiled.json")
 
 
-@pytest_asyncio.fixture(scope="module")
-async def account_with_validate_deploy_class_hash(
-    new_gateway_account_client: AccountClient,
+async def declare_account(
+    account: AccountClient, compiled_account_contract: str
 ) -> int:
     """
-    Returns a class_hash of the account_with_validate_deploy.cairo
+    Declares a specified account
     """
-    compiled_contract = Compiler(
-        contract_source=read_contract(
-            "account_with_validate_deploy.cairo", directory=CONTRACTS_DIR
-        ),
-        is_account_contract=True,
-    ).compile_contract()
 
-    declare_tx = await new_gateway_account_client.sign_declare_transaction(
-        compiled_contract=compiled_contract,
+    declare_tx = await account.sign_declare_transaction(
+        compiled_contract=compiled_account_contract,
         max_fee=MAX_FEE,
     )
-    resp = await new_gateway_account_client.declare(transaction=declare_tx)
-    await new_gateway_account_client.wait_for_tx(resp.transaction_hash)
+    resp = await account.declare(transaction=declare_tx)
+    await account.wait_for_tx(resp.transaction_hash)
 
     return resp.class_hash
 
 
+@pytest_asyncio.fixture(scope="module")
+async def account_with_validate_deploy_class_hash(
+    pre_deployed_account_with_validate_deploy: AccountClient,
+) -> int:
+    compiled_contract = read_contract("account_with_validate_deploy_compiled.json")
```

Comment:
Why is it not a constant like in the method below?

Reply:
Because we have a contract for the new account but do not have one for the old one. The old account is compiled and we keep it as a constant